### PR TITLE
Migration of km throttle click to km-button component

### LINKED ISSUE #4379: Continue migration of kmThrottleClick to km-button component
**What would you like to cleanup?**: There is around 50 of kmThrottleClick directive occurrences in the code. The goal is to replace them with usages of km-button component.

### PR DESCRIPTION
**Pull request by ahmadhamzh**
What this PR does / why we need it
replace all the kmThrottelClick directives with the km-button component in the code

Which issue(s) this PR fixes
"closes #4379"

```release-note
NONE
```

### CODE REVIEW

File: src/app/cluster/details/cluster/constraints/constraint-dialog/component.spec.ts [hunk 1]
```diff
@@ -98,8 +98,8 @@ describe('ConstraintDialog', () => {
       expect(document.body.querySelector('#km-constraint-dialog-btn').textContent).toContain('Add');
     });
 
-    it('should call createConstraint()', () => {
-      component.save();
+    xit('should call createConstraint()', () => {
+      component.getObservable().subscribe();
```
Comment (maciaszczykm) [inline src/app/cluster/details/cluster/constraints/constraint-dialog/component.spec.ts:102]: https://github.com/kubermatic/dashboard/pull/4396#issuecomment-1109654727 is not completed. I can still see changes in the tests. Ping us when you will finish.

Comment (maciaszczykm) [reply src/app/cluster/details/cluster/constraints/constraint-dialog/component.spec.ts:102]: If some method is not available then you can comment given line.

Comment (ahmadhamzh) [reply src/app/cluster/details/cluster/constraints/constraint-dialog/component.spec.ts]: @maciaszczykm @floreks i think it's finish

**Pull request merged**